util/predicates/cluster_predicates.go

```diff
@@ -218,6 +256,23 @@ func ClusterControlPlaneInitialized(scheme *runtime.Scheme, logger logr.Logger)
 	}
 }
```

Comment:
Can you please open a follow-up PR after the current PR is merged to deprecate:
* ClusterUnpausedAndInfrastructureReady with a note that ClusterPausedTransitionsOrInfrastructureReady should be used instead
* ClusterCreateInfraReady with a note that this predicate is never true so it doesn't make sense to use it

Reply:
Opened WIP PR: 
- #11300